tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/ServiceRecipientsSectionControllerTests.cs

```diff
@@ -0,0 +1,181 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using NHSD.BuyingCatalogue.Ordering.Api.Controllers;
+using NHSD.BuyingCatalogue.Ordering.Api.Models;
+using NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Builders;
+using NHSD.BuyingCatalogue.Ordering.Application.Persistence;
+using NHSD.BuyingCatalogue.Ordering.Domain;
+using NUnit.Framework;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Controllers
+{
+    [TestFixture]
+    [Parallelizable(ParallelScope.All)]
+    internal sealed class ServiceRecipientsSectionControllerTests
+    {
+        [Test]
+        public void Ctor_NullRepository_Throws()
+        {
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                var _ = new ServiceRecipientsSectionController(null, null);
+            });
+        }
+
+        [TestCase(null)]
+        [TestCase("INVALID")]
+        public async Task GetAllAsync_OrderDoesNotExist_ReturnsNotFound(string orderId)
+        {
+            var context = ServiceRecipientsTestContext.Setup();
+            context.Order = null;
+
+            var response = await context.Controller.GetAllAsync(orderId);
+            response.Should().BeEquivalentTo(new ActionResult<ServiceRecipientsModel>(new NotFoundResult()));
+        }
+
+        [Test]
+        public async Task GetAllAsync_OrganisationIdDoesNotMatch_ReturnsForbidden()
+        {
+            var context = ServiceRecipientsTestContext.Setup();
+            context.Order.OrganisationId = Guid.NewGuid();
+
+            var response = await context.Controller.GetAllAsync("myOrder");
+            response.Should().BeEquivalentTo(new ActionResult<ServiceRecipientsModel>(new ForbidResult()));
+        }
+
+        [Test]
+        public async Task GetAllAsync_NoServiceRecipient_ReturnsEmptyList()
+        {
+            var context = ServiceRecipientsTestContext.Setup();
+            var expected = new ServiceRecipientsModel
+            {
+                ServiceRecipients = new List<ServiceRecipientModel>()
+            };
+
+            var response = await context.Controller.GetAllAsync("myOrder");
+            response.Should().BeEquivalentTo(new ActionResult<ServiceRecipientsModel>(expected));
+        }
+
+        [Test]
+        public async Task GetAllAsync_SingleServiceRecipient_ReturnsTheRecipient()
+        {
+            var context = ServiceRecipientsTestContext.Setup();
+
+            const string orderId = "C0000014-01";
+
+            var serviceRecipients = new List<(ServiceRecipient serviceRecipient, ServiceRecipientModel expectedModel)>
+            {
+                CreateServiceRecipientData("ODS1", orderId)
+            };
+
+            context.ServiceRecipients = serviceRecipients.Select(x => x.serviceRecipient).ToList();
+
+            var expected = new ServiceRecipientsModel();

```

Comment:
Possible to use an object initializer here?

Reply:
👍